src/js/controller/controller.js

```diff
@@ -371,6 +370,7 @@ Object.assign(Controller.prototype, {
                     resolve(data);
                 });
                 loader.on(ERROR, function(error) {
+                    error.code = (error || 0) + ERROR_LOADING_PLAYLIST;
```

Comment:
if error is defined then we are setting `error + ERROR_LOADING_PLAYLIST`, not error.code
`error.code += ERROR_LOADING_PLAYLIST;`?